query.go

```diff
@@ -0,0 +1,78 @@
+package beacon
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/bigquery"
+)
+
+type Query struct {
+	refName string
+	allele  string
+	coord   *int64
+}
+
+func (q *Query) Execute(ctx context.Context, projectID, tableID string) (bool, error) {
+	query := fmt.Sprintf(`
+		SELECT count(v.reference_name) as count
+		FROM %s as v
+		WHERE %s
+		LIMIT 1`,
+		fmt.Sprintf("`%s`", tableID),
+		q.whereClause(),
+	)
+
+	bqClient, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return false, fmt.Errorf("creating bigquery client: %v", err)
+	}
+	it, err := bqClient.Query(query).Read(ctx)
+	if err != nil {
+		return false, fmt.Errorf("querying database: %v", err)
+	}
+
+	var result struct {
+		Count int
+	}
+	if err := it.Next(&result); err != nil {
+		return false, fmt.Errorf("reading query result: %v", err)
+	}
+	return result.Count > 0, nil
+}
+
+func (q *Query) ValidateInput() error {
```

Comment:
Since these are all required, you would be better to make a NewQuery(...) function that returns an error
func NewQuery(reference, allele string, coordinate int64) (*Query, error) {
 ...
}

Reply:
I ended up with this solution because the new api has 13 parameters and there are some conditions when one or the other is mandatory. I think it would be hard to maintain as an exported function